site-builder/src/site/resource.rs

```diff
@@ -458,3 +487,118 @@ pub(crate) fn full_path_to_resource_path(full_path: &Path, root: &Path) -> Resul
+                pattern: "test/*",
+                path: "test/file.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/foo.bar.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/foo-bar_baz.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "[invalid",
+                path: "file",
+                expected: false,
+            },
+        ];
+        for t in tests {
+            assert_eq!(
+                ResourceManager::is_pattern_match(t.pattern, t.path),
+                t.expected
+            );
+        }
+    }
+
+    #[ignore = "The test depends on the file system containing a walrus binary.
+Until we find a way to mock the walrus binary, this test will be ignored."]
+    #[tokio::test]
+    async fn test_derive_http_headers() {
+        let resource_manager = setup_resource_manager_mock().await;
+        let test_paths = vec![
+            // This is the longest path. So `/foo/bar/baz/*.svg` would persist over `*.svg`.
+            ("/foo/bar/baz/image.svg", "etag"),
+            // This will only match `*.svg`.
+            (
+                "/very_long_name_that_should_not_be_matched.svg",
+                "cache-control",
+            ),
+        ];
+        for (path, expected) in test_paths {
+            let result = resource_manager.derive_http_headers(path);
+            assert_eq!(result.len(), 1);
+            assert!(result.contains_key(expected));
+        }
+    }
+
+    /// Sets up a mock resource manager with the given headers configuration and resource path.
+    ///
+    /// Helper function for testing the `derive_http_headers` method.
+    async fn setup_resource_manager_mock() -> ResourceManager {
+        let walrus_mock = Walrus::new("walrus".to_string(), 1234, None, None, None);
+        // Define the headers configuration for mocking the resource manager.
+        let mut headers: BTreeMap<String, HttpHeaders> = BTreeMap::new();
+        headers.insert(
+            "/*.svg".to_string(),
+            HttpHeaders(BTreeMap::from([(
+                "cache-control".to_string(),
+                "public, max-age=86400".to_string(),
+            )])),
+        );
+        headers.insert(
+            "/foo/bar/baz/*.svg".to_string(),
+            HttpHeaders(BTreeMap::from([(
+                "etag".to_string(),
+                "\"abc123\"".to_string(),
+            )])),
+        );
+        let ws_resources = Some(WSResources {
+            headers: Some(headers),
+            routes: None,
+        });
+        ResourceManager::new(
+            walrus_mock.clone(),
+            ws_resources,
+            Some(PathBuf::from(&"mock/path/to/ws-resources")),
+        )
+        .await
+        .unwrap()
+    }
```

Comment:
Hint: the function `derive_http_headers` only requires the `ws_resources` and the `resource_path`, and therefore should not require this complicated mock to test.  

Suggestion: make `derive_http_headers` a static function, that takes the two inputs above. Then, add a method to `ResourceManager` that calls that function. This way you can test the static function without the need to mock the manager.

Reply:
Nice idea, I avoided it initially because I didn't want to change the public interface of the ResourceManager but now that I think more about it, it makes sense. 💯

Reply:
https://github.com/MystenLabs/walrus-sites/pull/272#:~:text=action%20skip%20ignored-,6510a9d,-Refactor%3A%20mock%20headers